src/machine/usb.go

```diff
@@ -302,11 +302,15 @@ func EnableMIDI(txHandler func(), rxHandler func([]byte), setupHandler func(usb.
 
 // EnableJoystick enables HID. This function must be executed from the init().
 func EnableJoystick(txHandler func(), rxHandler func([]byte), setupHandler func(usb.Setup) bool, hidDesc []byte) {
-	idx := bytes.Index(usb.DescriptorCDCJoystick.Configuration, []byte{
-		0x09, 0x21, 0x11, 0x01, 0x00, 0x01, 0x22,
-	})
-	binary.LittleEndian.PutUint16(usb.DescriptorCDCJoystick.Configuration[idx+7:idx+9], uint16(len(hidDesc)))
-	usb.DescriptorCDCJoystick.HID[2] = hidDesc
+	class, err := descriptor.FindClassHIDType(descriptor.CDCJoystick.Configuration, descriptor.ClassHIDJoystick.Bytes())
```

Comment:
EnableJoystick has been called twice, and the second call seems to fail to find it because ClassLength has already been set.
Why it is being called twice requires a little more checking.